feat(patcher): adding patcher library to readme file

Jacobbrewer1 opened the pull request:
## We want to ensure high quality of the packages. Make sure that you've checked the boxes below before sending a pull request.

- [x] I have read the [Contribution Guidelines](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#contribution-guidelines)
- [x] I have read the [Maintainers Note](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#maintainers)
- [x] I have read the [Quality Standards](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#quality-standards)

_Not every repository (project) will require every option, but most projects should. Check the Contribution Guidelines for details._

- [x] The repo documentation has a pkg.go.dev link.
- [x] The repo documentation has a coverage service link.
- [x] The repo documentation has a goreportcard link.
- [x] The repo has a version-numbered release and a go.mod file.
- [x] The repo has a continuous integration process that automatically runs tests that must pass before new pull requests are merged.
- [x] Continuous integration is used to attempt to catch issues prior to releasing this package to end-users.

## Please provide some links to your package to ease the review

- [x] forge link (github.com, gitlab.com, etc): [link](https://github.com/Jacobbrewer1/patcher)
- [x] pkg.go.dev: [link](https://pkg.go.dev/github.com/jacobbrewer1/patcher)
- [x] goreportcard.com: [link](https://goreportcard.com/report/github.com/jacobbrewer1/patcher)
- [x] coverage service link ([codecov](https://codecov.io/), [coveralls](https://coveralls.io/), etc.): [link](https://github.com/Jacobbrewer1/patcher/pull/93)

## Pull Request content

- [x] The package has been added to the list in alphabetical order.
- [x] The package has an appropriate description with correct grammar.
- [x] As far as I know, the package has not been listed here before.

## Category quality

_Note that new categories can be added only when there are 3 packages or more._

Packages added a long time ago might not meet the current guidelines anymore. It would be very helpful if you could check 3-5 packages above and below your submission to ensure that they also still meet the Quality Standards.

Please delete one of the following lines:

- [x] The packages around my addition still meet the Quality Standards.

Thanks for your PR, you're awesome! :sunglasses:

phanirithvij commented:
Please follow the pr template.

Jacobbrewer1 commented:
> Please follow the pr template.

Updated, I think. Thanks :smile:

phanirithvij commented:
If you don't mind answering

https://github.com/search?q=https%3A%2F%2Fx%3A%24%7BGITHUB_API_TOKEN%7D%40github.com&type=code&p=2

```
      - name: "Configure git for private modules"
        env:
          GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: git config --global url."https://x:${GITHUB_API_TOKEN}@github.com".insteadOf "https://github.com"
```

is this necessary at all? what private modules is patcher using exactly?

also your tests are not tracking code coverage, and as per awesome-go quality standard, an 80% code coverage is required.

Jacobbrewer1 commented:
> If you don't mind answering
> 
> https://github.com/search?q=https%3A%2F%2Fx%3A%24%7BGITHUB_API_TOKEN%7D%40github.com&type=code&p=2
> 
> ```
>       - name: "Configure git for private modules"
>         env:
>           GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
>         run: git config --global url."https://x:${GITHUB_API_TOKEN}@github.com".insteadOf "https://github.com"
> ```
> 
> is this necessary at all? what private modules is patcher using exactly?
> 
> also your tests are not tracking code coverage, and as per awesome-go quality standard, an 80% code coverage is required.

Thanks @phanirithvij 

I have removed those lines of code, they are from my template repo. I have also added checks for the test coverage as requested [here](https://github.com/Jacobbrewer1/patcher/pull/93).
